app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServerFactoryImpl.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.drone
+
+import io.mavsdk.System
+import io.mavsdk.mavsdkserver.MavsdkServer
+
+class DroneServerFactoryImpl: DroneServerFactory {
+
+    override fun createSimulation(simIP: String, simPort: String): DroneServerFactory.InstanceContainer {
+        val mavsdkServer = MavsdkServer()
+        val cloudSimIPAndPort = "tcp://${simIP}:${simPort}"
+        val mavsdkServerPort = mavsdkServer.run(cloudSimIPAndPort)
+        return DroneServerFactory.InstanceContainer(mavsdkServer, System("localhost", mavsdkServerPort))
+    }
+
+    override fun createDrone(): DroneServerFactory.InstanceContainer {
```

Comment:
Does this really provide a correct drone instance?

Reply:
Yes, from my understanding the instance is correct until the server is stopped